src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -0,0 +1,1008 @@
+      /* input */ "transformer_decoding_2_conv",
+      /* num_channels */ 32,
+      /* epsilon */ 1e-5);
+
+  nn_spec->add_multiplication(
+      /* name */ "transformer_instancenorm4__fwd_mult_gamma",
+      /* inputs */ {"transformer_instancenorm4__fwd_bn_",
+                    "transformer_decoding_2_inst_gamma"});
+
+  nn_spec->add_addition(
+      /* name */ "transformer_instancenorm4__fwd",
+      /* inputs */ {"transformer_instancenorm4__fwd_mult_gamma",
+                    "transformer_decoding_2_inst_beta"});
+
+  nn_spec->add_relu(
+      /* name */ "transformer_activation4",
+      /* input */ "transformer_instancenorm4__fwd");
+
+  nn_spec->add_padding(
+      /* name */ "transformer_pad5",
+      /* input */ "transformer_activation4",
+      /* padding_top */ 4,
+      /* padding_bottom */ 4,
+      /* padding_left */ 4,
+      /* padding_right */ 4,
+      /* padding_right */ padding_type::REFLECTIVE);
+
+  nn_spec->add_convolution(
+      /* name */ "transformer_conv5",
+      /* input */ "transformer_pad5",
+      /* num_output_channels */ 3,
+      /* num_kernel_channels */ 32,
+      /* kernel_height */ 9,
+      /* kernel_width */ 9,
+      /* stride_height */ 1,
+      /* stride_width */ 1,
+      /* padding */ padding_type::VALID,
+      /* weight_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_inner_product(
+      /* name */ "transformer_instancenorm5_gamma",
+      /* input */ "index",
+      /* num_output_channels */ 3,
+      /* num_input_channels */ num_styles,
+      /* weight_init_fn */ ones_weight_initializer(),
+      /* bias_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_inner_product(
+      /* name */ "transformer_instancenorm5_beta",
+      /* input */ "index",
+      /* num_output_channels */ 3,
+      /* num_input_channels */ num_styles,
+      /* weight_init_fn */ zero_weight_initializer(),
+      /* bias_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_instancenorm(
+      /* name */ "transformer_instancenorm5__fwd_bn_",
+      /* input */ "transformer_conv5",
+      /* num_channels */ 3,
+      /* epsilon */ 1e-5);
+
+  nn_spec->add_multiplication(
+      /* name */ "transformer_instancenorm5__fwd_mult_gamma",
+      /* inputs */ {"transformer_instancenorm5__fwd_bn_",
+                    "transformer_instancenorm5_gamma"});
+
+  nn_spec->add_addition(
+      /* name */ "transformer_instancenorm5__fwd",
+      /* inputs */ {"transformer_instancenorm5__fwd_mult_gamma",
+                    "transformer_instancenorm5_beta"});
+
+  nn_spec->add_sigmoid(
+      /* name */ "transformer_activation5",
+      /* input */ "transformer_instancenorm5__fwd");
+
+  nn_spec->add_scale(
+      /* name */ "stylizedImage",
+      /* input */ "transformer_activation5",
+      /* shape_c_h_w */ {1},
+      /* weight_init_fn */ const_weight_initializer(255.0));
+}
+
+void load_weights(std::unique_ptr<model_spec>& nn_spec, std::string& path) {
```

Comment:
Let's just make the first parameter have type `model_spec*`. No need to involve `unique_ptr` here

Reply:
Alright! No memory leakage here because of it?

Reply:
Passing a pointer in and of itself is memory-allocation-neutral. Using a raw pointer makes it more clear that ownership of the memory is not changing; using a `unique_ptr` leaves open the possibility that the function can delete the memory.